corehq/apps/users/static/users/js/roles.js

```diff
@@ -192,9 +193,37 @@ hqDefine('users/js/roles',[
             },
         };
 
+        self.allowSubmit = ko.computed(function () {
+            if (!self.roleBeingEdited()) {
+                return false;
+            }
+            if (!self.roleBeingEdited().name()) {
+                return false;
+            }
+            return true;
+        });
+
+        self.submitNewRole = function () {
```

Comment:
Don't you need to return false to keep this from reloading the page, now that you're genuinely submitting a form?

Reply:
Ethan showed me this: https://knockoutjs.com/documentation/submit-binding.html

So it seems like with knockout, it's not necessary to explicitly declare `return false`.

Reply:
nice, thanks for explaining